plugin/src/main/java/org/opensearch/ml/processor/MLInferenceIngestProcessor.java

```diff
@@ -0,0 +1,253 @@
+        }
+    }
+
+    private String getFieldPath(IngestDocument ingestDocument, String originalFieldName) {
+        TemplateScript.Factory originalField = ConfigurationUtils.compileTemplate(TYPE, tag, "field", originalFieldName, scriptService);
+        String originalFieldPath = ingestDocument.renderTemplate(originalField);
+        final boolean fieldPathIsNullOrEmpty = Strings.isNullOrEmpty(originalFieldPath);
+        if (fieldPathIsNullOrEmpty || !ingestDocument.hasField(originalFieldPath, true)) {
+            if (ignoreMissing) {
+                return null;
+            } else if (fieldPathIsNullOrEmpty) {
+                throw new IllegalArgumentException("field name in input_map [ " + originalFieldPath + "] cannot be null nor empty");
+            } else {
+                throw new IllegalArgumentException("field name in input_map: [" + originalFieldPath + "] doesn't exist");
+            }
+        }
+        return originalFieldPath;
+    }
+
+    private void appendFieldValue(
+        ModelTensorOutput modelTensorOutput,
+        String originalModelOutputFieldName,
+        String newModelOutputFieldName,
+        IngestDocument ingestDocument
+    ) {
+        Object modelOutputValue = null;
+        try {
+            modelOutputValue = getModelOutputField(modelTensorOutput, originalModelOutputFieldName, ignoreMissing);
+        } catch (IOException e) {
+            if (!ignoreMissing) {
+                throw new IllegalArgumentException("model inference output can not find field name: " + originalModelOutputFieldName, e);
+            }
+        }
+        ValueSource ingestValue = ValueSource.wrap(modelOutputValue, scriptService);
+        TemplateScript.Factory ingestField = ConfigurationUtils
+            .compileTemplate(TYPE, tag, newModelOutputFieldName, newModelOutputFieldName, scriptService);
+        ingestDocument.appendFieldValue(ingestField, ingestValue, false);
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static class Factory implements Processor.Factory {
+
+        private final ScriptService scriptService;
+        private final Client client;
+
+        public Factory(ScriptService scriptService, Client client) {
+            this.scriptService = scriptService;
+            this.client = client;
+        }
+
+        @Override
+        public MLInferenceIngestProcessor create(
+            Map<String, Processor.Factory> registry,
+            String processorTag,
+            String description,
+            Map<String, Object> config
+
+        ) throws Exception {
+            String model_id = ConfigurationUtils.readStringProperty(TYPE, processorTag, config, MODEL_ID);
+            Map<String, Object> model_config_input = ConfigurationUtils.readOptionalMap(TYPE, processorTag, config, MODEL_CONFIG);
+            List<Map<String, String>> input_map = ConfigurationUtils.readOptionalList(TYPE, processorTag, config, INPUT_MAP);
```

Comment:
Change to camel case and add `s` since this is list of map

Reply:
changed all attributes to camelcase. please review again